diesel/src/associations/belongs_to.rs

```diff
@@ -58,21 +60,21 @@ impl<'a, Parent, Child> BelongingToDsl<&'a Parent> for Child where
 }
 
 impl<'a, Parent, Child> BelongingToDsl<&'a [Parent]> for Child where
-    Parent: Identifiable,
+    &'a Parent: Identifiable,
     Child: HasTable + BelongsTo<Parent>,
-    Vec<&'a Parent::Id>: AsInExpression<<Child::ForeignKeyColumn as Expression>::SqlType>,
-    <Child as HasTable>::Table: FilterDsl<EqAny<Child::ForeignKeyColumn, Vec<&'a Parent::Id>>>,
+    Vec<Id<&'a Parent>>: AsInExpression<<Child::ForeignKeyColumn as Expression>::SqlType>,
+    <Child as HasTable>::Table: FilterDsl<EqAny<Child::ForeignKeyColumn, Vec<Id<&'a Parent>>>>,
```

Comment:
Funny idea for teaching: Tell people how to read a clause like this as an English sentence. E.g.:

> A [reference to a] Parent is Identifiable, i.e. we can find the corresponding row.

> We know about Child that it has a table and BelongsTo Parent.

> The [List] of [references to] Parent Ids can be turned into an `IN` expression based on the SqlType of the Child's ForeignKeyColumn.

> The Child's Table can be Filtered [using our Dsl] to get the rows where the Child's ForeignKeyColumn is Eq to Any of the entries in the [List] of Parent Ids.